src/main/java/org/ibp/api/java/impl/middleware/study/validator/ObservationImportRequestValidator.java

```diff
@@ -138,12 +138,84 @@ public BindingResult pruneObservationsInvalidForImport(final List<ObservationDto
 		return this.errors;
 	}
 
-	private boolean isObservationAlreadyExisting(final ObservationDto dto,
-		final Map<String, Map<String, ObservationDto>> existingObservationsMap, final Integer index) {
-		if (existingObservationsMap.containsKey(dto.getObservationUnitDbId())
-			&& existingObservationsMap.get(dto.getObservationUnitDbId()).containsKey(dto.getObservationVariableDbId())) {
-			this.errors.reject("observation.import.already.existing",
-				new String[] {index.toString(), dto.getObservationUnitDbId(), dto.getObservationVariableDbId()}, "");
+	public BindingResult pruneObservationsInvalidForUpdate(final List<ObservationDto> observationDtos) {
```

Comment:
Create ObservationUpdateRequestValidator and move the related methods for update validation there.